### ISSUE TITLE
[Retail Bug Report] Bags.lua:1850: attempt to call upvalue 'SortBags' (a nil value)

### ISSUE THREAD

**Issue by jeremy-hill** [tukui-org/ElvUI#576]
### What is the issue you are having?

Click 'Sort bags' button, receive below when /edebug on is set: 

Message: Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1850: attempt to call upvalue 'SortBags' (a nil value)
Time: Wed Nov 16 01:15:23 2022
Count: 1
Stack: Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1850: attempt to call upvalue 'SortBags' (a nil value)
[string "=[C]"]: ?
[string "@Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua"]:1850: in function <Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1848>

Locals: 

### What is the expected behavior?

Bags should sort

### What actually happened?

Lua error is thrown

### Suggested Solution/Workaround

Revert bag sort function and/or reimplement function

### Errors

```shell
Message: Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1850: attempt to call upvalue 'SortBags' (a nil value)
Time: Wed Nov 16 01:15:23 2022
Count: 1
Stack: Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1850: attempt to call upvalue 'SortBags' (a nil value)
[string "=[C]"]: ?
[string "@Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua"]:1850: in function <Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1848>

Locals:
```

### ElvUI Changelog & Troubleshooting

- [X] I have confirmed that I have read the [Changelog](https://www.tukui.org/download.php?ui=elvui&changelog).
- [X] I have verfied ElvUI is up to date by checking /estatus.
- [X] I have verfied my issue doesn't exist in the ElvUI Issue Tracker.
- [ ] I have not read step six and I am just clicking all boxes.
- [X] I have verfied my issue doesn't exist in the default UI.
- [X] I have disabled all addons except the three called 'ElvUI', 'ElvUI Libraries', and 'ElvUI Options' by using `/edebug on`? This is important in order to rule out that any of your other addons are causing the issue. If you manage to pinpoint a conflict between ElvUI and another addon, make sure you include that information.

**Comment by jeremy-hill** [comment]
Update: I think the Lua error is thrown if 'Use Blizzard Cleanup' is enabled.

**Comment by Nerothos** [comment]
Hope I can post this here, I get the same error when I try to sort reagent bag

```
Message: Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1794: attempt to call upvalue 'SortReagentBankBags' (a nil value)
Time: Wed Nov 16 17:03:46 2022
Count: 1
Stack: Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1794: attempt to call upvalue 'SortReagentBankBags' (a nil value)
[string "=[C]"]: ?
[string "@Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua"]:1794: in function <Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1784>

Locals: 
```

**Comment by NoNick96** [comment]
Here's my, full error:

```
33x Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1850: attempt to call upvalue 'SortBags' (a nil value)
[string "@Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua"]:1850: in function <Interface/AddOns/ElvUI/Core/Modules/Bags/Bags.lua:1848>

Locals:
(*temporary) = nil
(*temporary) = "attempt to call upvalue 'SortBags' (a nil value)"
E = <table> {
 GetTimeInfo = <function> defined @Interface/AddOns/ElvUI/Core/General/Math.lua:379
 mylevel = 60
 Bags = <table> {
 }
 privateVars = <table> {
 }
 modules = <table> {
 }
 FormatMoney = <function> defined @Interface/AddOns/ElvUI/Core/General/Math.lua:436
 StaticPopup_EscapePressed = <function> defined @Interface/AddOns/ElvUI/Core/General/StaticPopups.lua:432
 UpdateUnitFrames = <function> defined @Interface/AddOns/ElvUI/Core/General/Core.lua:1403
 SetMoversPositions = <function> defined @Interface/AddOns/ElvUI/Core/General/Movers.lua:455
 ScheduleTimer = <function> defined @Interface/AddOns/BagSync/libs/AceTimer-3.0/AceTimer-3.0.lua:94
 PriestColors = <table> {
 }
 RegisterModule = <function> defined @Interface/AddOns/ElvUI/Core/General/Core.lua:1813
 Cooldown_Options = <function> defined @Interface/AddOns/ElvUI/Core/General/Cooldowns.lua:129
 OnPauseCooldown = <function> defined @Interface/AddOns/ElvUI/Core/General/Cooldowns.lua:237
 UIScale = <function> defined @Interface/AddOns/ElvUI/Core/General/PixelPerfect.lua:55
 db = <table> {
 }
 SetEnabledState = <function> defined @Interface/AddOns/BagSync/libs/AceAddon-3.0/AceAddon-3.0.lua:420
 Layout = <table> {
 }
 UnregisterEventForObject = <function> defined @Interface/AddOns/ElvUI/Core/General/Core.lua:1718
 Distributor = <table> {
 }
 DBConvertSL = <function> defined @Interface/AddOns/ElvUI/Core/General/Core.lua:1234
 MountText = <table> {
 }
 INCOMPATIBLE_ADDONS = <table> {
 }
 UnregisterAllEvents = <function> defined @Interface/AddOns/BagSync/libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:169
 UIFrameFade = <function> defined @Interface/AddOns/ElvUI/Core/General/Animation.lua:278
 TimeIndicatorColors = <table> {
 }
 TexturePath = "ElvUI\Media\Textures\"
 StaticPopup_CreateSecureButton = <function> defined @Interface/AddOns/ElvUI/Core/General/StaticPopups.lua:1096
 Truncate = <function> defined @Interface/AddOns/ElvUI/Core/General/Math.lua:142
 TimeThreshold = 3
 ToggleMoveMode = <function> defined @Interface/AddOns/ElvUI/Core/General/Config.lua:69
 RemoveTableDuplicates = <function> defined @Interface/AddOns/ElvUI/Core/General/Core.lua:627
 WorldMap = <table> {
 }
 Disable = <function> defined @Interface/AddOns/BagSync/libs/AceAddon-3.0/AceAddon-3.0.lua:310
 DF = <table> {
 }
 SlideOut = <function> defined @Interface/AddOns/ElvUI/Core/General/Animation.lua:211
 Unhook = <function> defined @Interface/AddOns/ElvUI_Libraries/Core/Ace3/AceHook-3.0/AceHook-3.0.lua:417
 XPIsTrialMax = <function> defined @Interface/AddOns/ElvUI/Core/General/API.lua:512
 UnregisterObjectForVehicleLock = <function> defined @Interface/AddOns/ElvUI/Core/General/API.lua:403
 EscapeString = <function> defined @Interface/AddOns/ElvUI/Core/init.lua:164
 ObjectEventFrame = Frame {
 }
 StaticPopup_Resize = <function> defined @Interface/AddOns/ElvUI/Core/General/StaticPopups.lua:697
 Tutorials = <function> defined @Interface/AddOns/ElvUI/Core/General/Tutorials.lua:110
 SetDefaultModuleLibraries = <function> defined @Interface/AddOns/BagSync/libs/AceAddon-3.0/AceAddon-3.0.lua:363
 ProfileTableToPluginFormat = <function> defined @Interface/AddOns/ElvUI/Core/General/Core.lua:830
 ShortValueDec = "%.1f"
 oUF = <table> {
 }
 unitFrameElements = <table> {
 }
 Shake = <function> defined @Interface/AddOns/ElvUI/Core/General/Animation.lua:154
 Config_CreateBottomButtons = <function> defined @Interface/AddOns/ElvUI/Core/General/Config.lua:957
 RawHookScript = <function> defined @Interface/AddOns/ElvUI_Libraries/Core/Ace3/AceHook-3.0/AceHook-3.0.lua:394
 Classic = false
 BuildPrefixValues = <function> defined @Interface/AddOns/ElvUI/Core/General/Math.lua:34
 myrole = "DAMAGER"
 UnregisterAllEventsForObject = <function> defined @Interface/AddOns/ElvUI/Core/General/Core.lua:1745
 SetupComplete = <function> defined @Interface/AddOns/ElvUI/Core/General/Install.lua:522
 PreviousPage = <function> defined @Interface/AddOns/ElvUI/Core/General/Install.lua:764
 XPIsUserDisabled = <function> defined @Interf
```

**Comment by bunjiboys** [comment]
Looks like the API changed in 10.0.2, and its now using a new namespace for the function. If you open `ElvUI/Core/Modules/Bags/Bags.lua` and change line 1850 from
```lua
SortBags()
```
to
```lua
C_Container.SortBags()
```

The entire function would thus look like below
```lua
		--Sort Button
		f.sortButton:SetScript('OnClick', function()
			if E.Retail and B.db.useBlizzardCleanup then
				C_Container.SortBags()
			else
				f:UnregisterAllEvents() --Unregister to prevent unnecessary updates
				if not f.sortingSlots then B:SortingFadeBags(f, true) end
				B:CommandDecorator(B.SortBags, 'bags')()
			end
		end)
```

**Comment by bunjiboys** [comment]
Yeah I wasn't sure at first, cause I didn't notice the grouping of all the other `C_Container` methods, but refactored to use the approach with local variables, and since all the other `C_Container` local variables appear to be grouped together, I moved these definitions to be grouped with all the other `C_Container` related variables

**Comment by kodewdle** [comment]
fixed by f7ab01d54d thanks
